fe/fe-core/src/main/java/org/apache/doris/alter/IndexChangeJob.java

```diff
@@ -370,17 +372,22 @@ protected void runRunningJob() throws AlterCancelException {
             LOG.info("inverted index tasks not finished. job: {}, partitionId: {}", jobId, partitionId);
             List<AgentTask> tasks = invertedIndexBatchTask.getUnfinishedTasks(2000);
             for (AgentTask task : tasks) {
-                if (task.getFailedTimes() > 3) {
+                if (task.getFailedTimes() >= MAX_FAILED_NUM) {
                     LOG.warn("alter inverted index task failed: " + task.getErrorMsg());
-                    List<Long> failedBackends = failedTabletBackends.computeIfAbsent(task.getTabletId(),
-                            k -> Lists.newArrayList());
+                    // If error is E-216, it indicates obtaining lock failed.
+                    // we should retry this task.
+                    if (task.getErrorMsg().contains("E-216")) {
```

Comment:
it is not robust to do string contains here. Is there any other way ? Error code etc.